signingscript/src/signingscript/rcodesign.py

```diff
@@ -100,12 +101,33 @@ async def rcodesign_notary_wait(submission_id, creds_path):
         submission_id,
     ]
     log.info(f"Polling Apple Notary service for notarization status. Submission ID {submission_id}")
-    exitcode, _ = await _execute_command(command)
+    exitcode, logs = await _execute_command(command)
     if exitcode > 0:
         raise RCodesignError(f"Error polling notary service. Exit code {exitcode}")
+
+    await rcodesign_check_result(logs)
     return
 
 
+async def rcodesign_check_result(logs):
+    """Checks notarization results
+    rcodesign cli call can exit with 0 even though the notarization has failed
+    Args:
+        logs (list<str>): output from polling result
+    """
+    re_inprogress = re.compile("^poll state after.*InProgress$")
+    re_accepted = re.compile("^poll state after.*Accepted$")
+    re_invalid = re.compile("^poll state after.*Invalid$")
```

Comment:
For some reason `rcodesign` doesn't seem to error (exitcode > 0) when the package is invalid.

Reply:
Did you file an issue against rcodesign that could be referenced in a comment here?

Reply:
I was debating it.
The command used here was `notary-wait` which technically "waited" successfully, the fact that the logs say the package is invalid might not warrant an exit code from rcodesign itself.

Reply:
Well I think it should :)